.github/workflows/build-test-deploy.yml

```diff
@@ -1014,63 +1014,94 @@ jobs:
           #     raise Exception(
           # Exception: could not find specified deployment in state file ‘/tmp/nix-shell.QuNDXa/tmpd0tvopatnixops/state.nixops’          
           # nix develop .#sre --show-trace -L --command nixops deploy --dry-run --debug --keep-going --show-trace
-  deploy-picasso-frontend:
-    name: "Firebase Picasso Web App Deploy"
-    needs:
-      - local-integration-tests
-      - unit-tests
-    runs-on: ubuntu-latest
+
+
+  frontend-nix-deploy-firebase:
+    name: "Firebase Frontend App Deploy"
+    needs: common-deps
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: packagecomposablenode-${{ github.ref }}
+      cancel-in-progress: true
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
+      - name: Check if hash changes
+        if: always()
+        id: feStaticHashChanges
+        run: |
+          if [ `nix build --dry-run "github:ComposableFi/composable#frontend-static" --json` == `nix build --dry-run ".#frontend-static" --json` ]; then
+            echo "::set-output name=hashChange::false"; 
+          else echo "::set-output name=hashChange::true"; 
+          fi
+
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        if: steps.feStaticHashChanges.outputs.hashChange == 'true'
```

Comment:
Can't we remove this entire step, and instead embed it in the `if:` of the deploy step?

Reply:
Agree this doesn't look pretty, but I was unsuccessful to signal skipping the job from inside the bash command, exiting with non-zero code would make the build cancel with error.

Reply:
Alright! At least functionally equivalent for now. At some point we could perhaps add a bash script to the repo which emits a boolean, so we can do 

```
if: ./scripts/changed github:ComposableFi/composable#frontend-static .#frontend-static
```